apps/aecore/src/aec_trees.erl

```diff
@@ -249,6 +250,94 @@ contracts(Trees) ->
 set_contracts(Trees, Contracts) ->
     Trees#trees{contracts = Contracts}.
 
+-spec sum_total_coin(trees()) -> #{ 'accounts'         => non_neg_integer()
+                                  , 'contracts'        => non_neg_integer()
+                                  , 'contract_oracles' => non_neg_integer()
+                                  , 'oracles'          => non_neg_integer()
+                                  , 'oracle_queries'   => non_neg_integer()
+                                  , 'locked'           => non_neg_integer()
+                                  }.
+sum_total_coin(Trees) ->
+    Accounts  = accounts(Trees),
+    Channels  = channels(Trees),
+    Contracts = contracts(Trees),
+    Oracles   = oracles(Trees),
+    LockAccount   = aec_governance:locked_coins_holder_account(),
+    AIterator     = aec_accounts_trees:mtree_iterator(Accounts),
+    FirstAccount  = aeu_mtrees:iterator_next(AIterator),
+    CIterator     = aesc_state_tree:mtree_iterator(Channels),
+    FirstChannel  = aeu_mtrees:iterator_next(CIterator),
+    ChannelAmount = sum_channels(FirstChannel, 0),
+    Sum = #{ 'accounts'         => 0
+           , 'channels'         => ChannelAmount
+           , 'contracts'        => 0
+           , 'contract_oracles' => 0
+           , 'oracles'          => 0
+           , 'oracle_queries'   => 0
+           , 'locked'           => 0
+           },
+    sum_accounts(FirstAccount, LockAccount, Contracts, Oracles, Sum).
+
+sum_accounts('$end_of_table',_LockAccount,_Contracts,_Oracles, Acc) ->
+    Acc;
+sum_accounts({LockAccount, SerAccount, Iter}, LockAccount, Contracts, Oracles, Acc) ->
+    Account = aec_accounts:deserialize(LockAccount, SerAccount),
+    Balance = aec_accounts:balance(Account),
+    Acc1 = maps:update_with(locked, fun(X) -> X + Balance end, Acc),
+    Next = aeu_mtrees:iterator_next(Iter),
+    sum_accounts(Next, LockAccount, Contracts, Oracles, Acc1);
+sum_accounts({Pubkey, SerAccount, Iter}, LockAccount, Contracts, Oracles, Acc) ->
```

Comment:
I see double accounting for coins in `aec_governance:locked_coins_holder_account()` is prevented by previous clause. Ok.